packages/onboarding/src/cart/index.tsx

```diff
@@ -181,6 +184,8 @@ export const createSiteWithCart = async (
 	// 	return;
 	// }
 
+	const isOnboarding = isOnboardingFlow( flowName );
```

Comment:
Maybe we could remove this check and only check if site goals were passed?

Reply:
I'm inclined to keep this check in place. Maybe even reinforce it with a check of `isGoalsFirstExperiment`

[There are issues like this](https://github.com/Automattic/wp-calypso/issues/96481#issuecomment-2487492997) where sometimes there are phantom goals left in the store from previous bailed user attempts. I wouldn't want to set those old phantom goals that just happen to be in the store onto the site.

Reply:
In order to get access to the `isGoalsFirstExperiment` value, maybe it'd be easier to always pass the `site_goals`  if they're passed to `createSiteWithCart`. And then to do the `isOnboardingFlow && isGoalsFirstExperiment` up in the `create-site` step? You've got more easy access to selectors and hooks in the step component.